### DISCUSSION TITLE
MiniDSP 8x8 plugin IP auto discovery

### DISCUSSION THREAD

**Discussion by minimaded** [mrene/minidsp-rs#575] [General]
Has anyone gotten the MiniDSP 8x8 plugin to auto-discover the IP address of the minidsp-rs server?

I have added the following to "/etc/minidsp/config.toml"

`advertise = { ip = "192.168.1.181", name="MiniDSP 10x10HD" }`

The Android app detects the server and can control the connected MiniDSP 10x10HD fine.

Wireshark on my Windows PC shows the UDP broadcasts coming from the server but the plugin does not detect the IP address.

The plugin works fine if I manually enter the IP address.

Alternatively, does anyone know how to get the plugin to remember the IP address so I don't have to re-enter it every time I use the plugin?

Thanks

**Comment by minimaded** [comment]
Found the issue. I searched for "3999" (the UDP port) in the Wi-DG firmware and found this in the Alexa client.

// UDP
var udpSocket = require('dgram').createSocket('udp4');
udpSocket.on('message', function (d) {
  // If MiniDSP discovery packet
  if (d[0] == 0x80 && d[2] == 0x05 && d[3] == 0xA0) {
    var hwid = String(d[18]);
    var dspNo = String(d[21]);
    if (hwid != currentDevice.hwid) {
      console.log("Device Found: " + deviceTypes[hwid] + ' : ' + dspNo + '.');
      currentDevice.hwid = hwid;
      currentDevice.dspNo = dspNo;
    }
  }
});

I changed the first 4 bytes of the UDP broadcast to "0x80, 0x00, 0x05, 0xA0" and it was immediately recognized by the plugin.

Not sure where the change needs implementing in minidsp-rs as I am using an ESP32 as a USB to WiFi bridge.

**Reply by mrene** [reply]
Oh good find! I'll change the broadcast packet so it starts with a compatible header.
